test/date-time/date-time-create.test.ts

```diff
@@ -0,0 +1,1068 @@
+        then  it should throw a validation error`,
+            () =>
+            {
+                checkIsInvalid("20230229", validTimeCode);
+            }
+        );
+
+        test(`Given a valid time code(1000) and zone, and date code with invalid day feb 30 20240230
+        when DateTime is created from that codes
+        then  it should throw a validation error`,
+            () =>
+            {
+                checkIsInvalid("20240230", validTimeCode);
+            }
+        );
+
+        test(`Given a valid time code(1000) and zone, and date code feb 29 on leap year 20240229
+        when DateTime is created from that codes
+        then  it should return the DateTime object`,
+            () =>
+            {
+                Assert.ok(DateTime.createFromCodes("20240229", validTimeCode, "utc"));
+            }
+        );
+
+        test(`Given a valid time code(1000) and zone, and date code with invalid day 20240431 (April 31)
+        when DateTime is created from that codes
+        then  it should throw a validation error`,
+            () =>
+            {
+                checkIsInvalid("20240431", validTimeCode);
+            }
+        );
+
+        test(`Given a valid time code(1000) and zone, and date code with invalid month 0 20240001 
+        when DateTime is created from that codes
+        then  it should throw a validation error`,
+            () =>
+            {
+                checkIsInvalid("20240001", validTimeCode);
+            }
+        );
+
+        test(`Given a valid time code(1000) and zone, and date code with invalid month 13 20241301 
+        when DateTime is created from that codes
+        then  it should throw a validation error`,
+            () =>
+            {
+                checkIsInvalid("20241301", validTimeCode);
+            }
+        );
+
+        test(`Given a valid time code(1000) and zone, and date code with invalid year >9999 100000101 
+        when DateTime is created from that codes
+        then  it should throw a validation error`,
+            () =>
+            {
+                checkIsInvalid("100000101", validTimeCode);
+            }
+        );
+
+        test(`Given a valid date code(20240101), time code(1000) and zone as utc
+        when DateTime is created from that codes
+        then zone property should be utc`,
+            () =>
+            {
+                Assert.strictEqual(DateTime.createFromCodes(validDateCode, validTimeCode, "utc").zone, "utc");
+            }
+        );
+
+        test(`Given a valid date code(20240101), time code(1000) and zone as UTC+5:30
+        when DateTime is created from that codes
+        then zone property should be UTC+5:30`,
+            () =>
+            {
+                Assert.strictEqual(DateTime.createFromCodes(validDateCode, validTimeCode, "UTC+5:30").zone, "UTC+5:30");
+            }
+        );
+
+        test(`Given a valid date code(20240101), time code(1000) and zone as America/Los_Angeles
```

Comment:
Add test for the happy path, given this datecode and timecode values the expected datetime value is created